src/aind_slims_api/operations/ecephys_session.py

```diff
@@ -0,0 +1,158 @@
+"""
+Module defining operations to build EcephysSession.
+"""
+
+import logging
+from typing import List, Optional
+from pydantic import BaseModel
+from aind_slims_api import SlimsClient
+from aind_slims_api.exceptions import SlimsRecordNotFound
+from aind_slims_api.models.mouse import SlimsMouseContent
+from aind_slims_api.models.ecephys_session import (
+    SlimsMouseSessionResult,
+    SlimsStreamsResult,
+    SlimsStimulusEpochsResult,
+    SlimsExperimentRunStepContent,
+    SlimsExperimentRunStep,
+    SlimsDomeModuleRdrc,
+    SlimsRewardDeliveryRdrc,
+    SlimsRewardSpoutsRdrc,
+    SlimsGroupOfSessionsRunStep,
+    SlimsMouseSessionRunStep,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class EcephysSession(BaseModel):
+    """
+    Pydantic model encapsulating all session-related responses.
+    """
+
+    session_group: Optional[SlimsExperimentRunStep]
+    session_result: Optional[SlimsMouseSessionResult]
+    streams: Optional[List[SlimsStreamsResult]] = []
+    stream_modules: Optional[List[SlimsDomeModuleRdrc]] = []
+    reward_delivery: Optional[SlimsRewardDeliveryRdrc] = None
+    reward_spouts: Optional[SlimsRewardSpoutsRdrc] = None
+    stimulus_epochs: Optional[List[SlimsStimulusEpochsResult]] = []
+
+
+class SlimsEcephysSessionOperator:
```

Comment:
Do you see the need for more methods on this class in the future? If it's just fetching the collection, I would be inclined to have that just be a public function in the module. 
```python
class EcephysSession(BaseModel):
    session_group: Optional[SlimsExperimentRunStep]
    ...

def fetch_ecephys_session(slims_client: Optional[SlimsClient], subject_id:str) -> list[EcephysSession]:
    ...
```

If there's the potential for more operations you'd want to do with a session, then maybe I would merge the Operator class with the EcephysSession collection object above, so that it represents a Session concept with relevant attributes and methods.
```python
class SlimsEcephysSessionOperator:
    session_group: SlimsExperimentRunStep
    ...

    def __init__(self, slims_client: Optional[SlimsClient], subject_id: str, content_run_pk: int): ...

    def end_session(self): ...

    def add_step(self, ...): ...

def fetch_sessions(slims_client, subject_id) -> list[SlimsEcephysSessionOperator]:
```

I think I'm skeptical of having to instantiate an `operator` in addition to the `client` if it's not going to be a long-lasting helpful object. I know you wanted to have it inherit from SlimsClient which also solves that, though I think that has some drawbacks